synapse/config/server.py

```diff
@@ -321,9 +321,11 @@ def read_config(self, config, **kwargs):
                 self.presence_router_config,
             ) = load_module(presence_router_config, ("presence", "presence_router"))
 
-        # Whether to update the user directory or not. This should be set to
-        # false only if we are updating the user directory in a worker
-        self.update_user_directory = config.get("update_user_directory", True)
```

Comment:
Should we keep this config option for now (for backwards compatibility), but let it emit a warning if defined?

What should we do if it has a `False` value? Default to "synapse.app.user_dir"?

Reply:
I think with a `False` value, we should interpret it as "undefined", unless the local worker is the user_dir worker, then it'll suddenly "match", this should be enough to be backwards-compatible, but we should emit a compatibility warning when this happens.

Reply:
We would probably want to have some sort of backwards compatibility here, yes. We usually do that when updating the configuration.

Reply:
Alright, but do you think the backwards compatibility method described in my second comment looks alright?